### ISSUE TITLE
Alternate headwords for pw

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/PWK#106]
We tackle the task of generating alternate headwords for pw dictionary.

Preliminary outline of the approach:
* Filter entries based on the first line of data (the line after the metaline)
* Parse the implied headwords (based on the broken bar in that first line of entry)
  * recognize also hom and (?) roots
* Use this parse to construct k2 of the metaline base.  When there is more than one headword, this will result in a comma-separated list in k2
* construct parallel list of k1 from the list of k2.
* construct pw_hwextra.txt (for csl-orig) from the k1-k2 list
  * this will generate essentially duplicate entries in pw.xml for the extra 'alternate' headwords.

Note: no attempt to generate alternate headwords from upasargas of verb entries.

**Comment by funderburkjim** [comment]
## markup observations

* `<h>` appears in metalines in pw-main, but not in pw-vn. 
   * the present work will remove that `<h>N` and embed it into k2 (cf. GRA markup)
* `*`  - `29339 matches for "[*].*¦" ` . Boehtlingk. 
   * A word, a meaning, a construction or a gender that has so far only been 
     listed  by grammarians or lexicographers has been designated with *. [ref:](https://sanskrit-lexicon.uni-koeln.de/scans/csldev/csldoc/build/dictionaries/prefaces/pwpref/pwpref02.html)
* √  indicates a root . Andhrabharati addition. `3103 matches for "√{#.*¦"`
* √! indicates a 'denominative' root. Andhrabharati addition. `1294 matches for "!√{#.*¦"`

`√ and !` do not appear in k2 of metaline
`*` appears in k2 of metaline: `29069 matches for "<k2>[*]"`

**Comment by Andhrabharati** [comment]
@funderburkjim 

It appears that you are analyzing the pw file data, for marking the **prospective** alt. HWs yourself.

I had mentioned earlier (in #104) that **the header portion** may be looked at to get these words; but there appear to be more entries that contain alt. HWs (which I had missed before in my posted file that formed the base for the cdsl version).

Here are a few (43 no.s) such L-entries on a quick searching--

16678, 22390, 23281, 26293, 28410,
30597, 34090, 34556, 36315, 37930,
39355, 39831, 39852, 43702, 43761,
44078, 44931, 46971, 55339, 55378,
56092, 59172, 61689, 61801, 74511,
77163, 80864, 87626, 88417, 89051,
94286, 97985, 100189, 102651, 103657,
110117, 110192, 112382, 113429, 115433,
125488, 130735, 133969

**There could be more like these, and hope you would try to get those as well.**
`--------------------------------------------`
PS. You had tried to get to the <7k figure that I mentioned at #104, and suggested discarding the hom-tag entries in the process; but there are little over 200 entries having a hom-tag that contain alt. HWs. [**My earlier count was 6986.**]

**Comment by Andhrabharati** [comment]
Once you finish your exercise, I might be able to compare your file with my version (not posted so far) for any missings/changes.

**Comment by funderburkjim** [comment]
Yes, I am working on those alt-headwords now.  Will share my work with you for comparison, perhaps later this week.

**Comment by Andhrabharati** [comment]
@funderburkjim 

Now, I've looked for the 'missed' alt. HW entries in the erstwhile pwkvn file, and found 15 such.

```
<L>200221<pc>1-284-a<k1>ajagAva<k2>ajagAva
<L>203121<pc>2-299-c<k1>KaqIna<k2>KaqIna
<L>203284<pc>3-247-a<k1>aganDi<k2>aganDi
<L>203898<pc>3-253-c<k1>arTasaMhata<k2>arTasaMhata
<L>204099<pc>3-256-a<k1>AmravAwaka<k2>*AmravAwaka
<L>204416<pc>3-259-b<k1>kalvowaka<k2>kalvowaka
<L>204577<pc>3-261-a<k1>gArDapfzwa<k2>gArDapfzwa
<L>205639<pc>4-299-a<k1>cUlha<k2>cUlha
<L>206832<pc>5-249-c<k1>ikawI<k2>*ikawI
<L>207055<pc>5-252-a<k1>kikviwa<k2>kikviwa
<L>207313<pc>5-255-a<k1>jvalana<k2>jva/lana
<L>209005<pc>6-302-b<k1>devaniSrayaRI<k2>devaniSrayaRI
<L>209263<pc>6-305-b<k1>mAtaNgavedi<k2>*mAtaNgavedi
<L>213195<pc>7-312-a<k1>avimanas<k2>avimanas
<L>213697<pc>7-315-c<k1>asevana<k2>asevana
```

**Comment by funderburkjim** [comment]
## systematic additional headwords

My approach to determine NEW secondary headwords for an entry is based on
analysis of the 'broken-bar' line of the entry.

* 158370 is the  current count for pw.txt entries.
  * `grep -E '^<L>' ../temp_pw_2.txt | wc -l`
* 8444 entries are previously identified as multiple headwords 
  * `grep -E '{#.*?#}.*{#.*?#}.*¦' ../temp_pw_2.txt | wc -l`
* 3103 entries are identified as roots
  * `grep -E '√.*¦' ../temp_pw_2.txt | wc -l`

Excluding these two groups, there are approx. 31054 entries which *might*
  have multiple headwords.
`grep -E '^[^{√]*{#[^#]*#}[^{]*¦.*{#' ../temp_pw_2.txt | wc -l`

But there are many patterns in these 31054 entries which disqualify the
entry from implying extra headwords. For example, in `{#akulI#}¦ <ab>v. l.</ab> für {#aNkulI#}.`, `{#aNkulI#}` is not an extra headword. The pattern is `<ab>v. l.</ab> für {#`
RESTRICTED to cases with just one {#X#} after ¦

After excluding this and many other patterns, there remain 10788 candidates.
Certainly some of these should also be excluded; but at this point, the
exclusion-by-pattern approach has become unproductive.

Thus, the approach changes course to apply patterns to INCLUDE subsets of the
candidates. For such an included entry, there need to be changes to:
* the broken bar line ( ¦ moves after last sub-headword phrase {#X#} )
* the `<k2>` field of metaline.

The file [temp_change_3_01.txt](https://github.com/sanskrit-lexicon/PWK/files/14290650/temp_change_3_01.txt)
Shows the changes made for the pattern: `{#°tA#}` or `{#°tva#}` (is the only {#X#} after ¦ ).
There are 915 cases here.
Later pw.txt will be changed by applying the changes in this file, and similar changes for other patterns.  
Eventually, this approach should get *most* of the implied extra headwords.

@Andhrabharati  Before proceeding much further, I wanted your take on this approach.

**Comment by Andhrabharati** [comment]
I would suggest 'restricting' **this phase** to mark and bring-out the 'primary HWs' (as I termed them) [single or 'grouped' entries], that occur **at the beginning of the entry** in the printed lexicon.

We definitely need to bring other 'inner' HWs also, that occur in multiple ways, but this could be done in **another/next phase** - 
(a) in-line HWs [that are within braces in the running text]
(b) implied HWs [mostly with a preceding "also' etc.]
(c) indicative HWs [like the -tA, -tva varieities, that mostly do not have any 'objective' body, but just a mention of the word]
(d) variant form HWs [with a preceding "written", "v. l.", "w. r."  etc.]

And then, we should look for the composite/compound words that occur inside the body portion of the above entries, and suitably bring them out.

--------------------------
BTW, I see 158375 `<L>` entries, not 158370 as mentioned above, in the combined pw.txt.

**Comment by Andhrabharati** [comment]
My personal opinion is that we should mark these 'secondary' HWs with <div n="x" > style [as done in GRA], "x" covering various forms that we come across in the particular lexicon.

And then, list those various groups within the 'main' entry somewhere (like the separate althw file seen in some cdsl works), to come under the "search" criterion.

**Comment by funderburkjim** [comment]
> BTW, I see 158375 \<L> entries, not 158370

I merged 5 entries:
```
parvan L=96646 merge into 69945 pora
pravAla L=73144 merge into  73143, pravAqa
AzwakIya L=16300 merge into 16299 Azwaka
Dru L= 55950 merge into Dru 55949
peSI L=69764 merge into peSI 69763
```

**Comment by funderburkjim** [comment]
## additional denominative roots
```
475 matches for "^!√{#[^#]*y#}¦, {#°yat[ie]#}"  already marked
102 matches for "^{#[^#]*y#}¦, {#°yat[ie]#}"  add !√ markup
```

**Comment by Andhrabharati** [comment]
> I merged 5 entries:

Looked for other entries with similar pattern `¦\n<LEND`, and found that 5 entries (44904, 49112, 53788, 58874 and 113078) have missed the body portion.

**Comment by Andhrabharati** [comment]
> 102 matches for "^{#[^#]*y#}¦, {#°yat[ie]#}"  add !√ markup

All these occur in the erstwhile pwkvn portion; I seem to have skipped marking them.

BTW, found one entry L-45920, which has √ mark, but should be with the !√ mark; and it has a typo **tilakaya** for **tilakay**

**Comment by funderburkjim** [comment]
> other entries with similar pattern ¦\n<LEND

Those five had missing text in cdsl pw.  I have added the text.
[temp_missing_body_AB.txt](https://github.com/sanskrit-lexicon/PWK/files/14314921/temp_missing_body_AB.txt)

**Comment by funderburkjim** [comment]
> Probably there could be some places missing the √ mark

 I generated a list of possible missing √ mark entries by
search for headword pattern `consonant-root-consonant` 
```
regex without hom (python syntax)
r'^[*]?{#[kKgGNcCjJYwWqQRtTdDnpPbBmyrlvSzsh|][aAiIuUfFxXeEoOMH][kKgGNcCjJYwWqQRtTdDnpPbBmyrlvSzsh|]#}¦'

regex with hom:  
r'^<hom>[^<]*</hom> [*]?{#[kKgGNcCjJYwWqQRtTdDnpPbBmyrlvSzsh|][aAiIuUfFxXeEoOMH][kKgGNcCjJYwWqQRtTdDnpPbBmyrlvSzsh|]#}¦'
```
513 found.  
Manually examined all. 
* 358 need √ mark (251 of these are in the pwkvn entries)
* 155 don't need √ mark.


details: [temp_possible_roots_edit.txt](https://github.com/sanskrit-lexicon/PWK/files/14315248/temp_possible_roots_edit.txt)

@Andhrabharati  Do you agree that √ mark should be added (before ¦) for these 358?

**Comment by Andhrabharati** [comment]
Yes pl., somehow I had skipped these markings while at pwkvn.

**Comment by Andhrabharati** [comment]
> Now, I've looked for the 'missed' alt. HW entries in the erstwhile pwkvn file, and found 15 such.

Found another 8 entries in pwkvn portion, that come under this--
L-200048, L-200334, L-201819, L-206328, L-208193, L-209653, L-214672, L-221290

**Comment by Andhrabharati** [comment]
> @Andhrabharati Before proceeding much further, I wanted your take on this approach.

Just curious to know your conclusion on how to proceed further on the task, @funderburkjim !!

**Comment by funderburkjim** [comment]
## interim progress
I've marked the additional 358 roots.

> 'primary HWs' (as I termed them) [single or 'grouped' entries], that occur at the beginning of the entry in the printed lexicon'close' althws

I like this idea and am proceeding to see if I find any more in addition to those you have mentioned in above comments.

**Comment by Andhrabharati** [comment]
Glad to hear this, @funderburkjim !

Working with a 'common' thinking/process definitely makes the collaborative effort easier, facilitating the comparision (between the two works) quicker and fruitful.

**Comment by Andhrabharati** [comment]
I have many more entries that come under the alt.HW type & the 'root' type now.

**Comment by funderburkjim** [comment]
##  altheadwords 
[temp_change_1a_2_althws.txt](https://github.com/sanskrit-lexicon/PWK/files/14353496/temp_change_1a_2_althws.txt)

This file contains changes for alternate headwords from 2 sources:
* 66 that you have listed above
* 26 that I have added

@Andhrabharati  Please review and provide corrections as needed.

Note: I have used the GRA model for deriving extra entries for PW (pw_hwextra.txt) from `<k2>`.
Currently: 2516 extra headwords from 1589 `<k2>`s.

> many more entries that come under the alt.HW type & the 'root' type now.

If you provide these, I'll make changes for them.

**Comment by Andhrabharati** [comment]
@funderburkjim 

My file now contains **547 (main) and 663 (vn) lines** differing with the CDSL (combined) file, ignoring the meta-lines (as I did not populate the k2-field yet).

If you post your full file, I can do a diff with my file and list out the differing lines.

**Comment by funderburkjim** [comment]
[temp_pw_2.zip](https://github.com/sanskrit-lexicon/PWK/files/14362314/temp_pw_2.zip) my current version
[temp_change_pw_0_2.txt](https://github.com/sanskrit-lexicon/PWK/files/14362320/temp_change_pw_0_2.txt)  Changes from the current csl-orig pw.txt.  All changes thus far were done while keeping the number of lines the same.  This file shows the line-by-line diff.

Further details in the [pwkissues/issue106 folder](https://github.com/sanskrit-lexicon/PWK/tree/master/pwkissues/issue106).

**Comment by Andhrabharati** [comment]
Thank you @funderburkjim for the files.

Seen that just over 900 lines (616: main and 315: vn) are differing between our files.

Will go through them tomorrow and after necessary corrections (if necessary) in my file, shall post the differing lines for your persual and further action.

**Comment by Andhrabharati** [comment]
Here are the files that I had made--
1. separated the VN data from the combined CDSL file: [pwkvn_2 (CDSL).txt](https://github.com/sanskrit-lexicon/PWK/files/14372254/pwkvn_2.CDSL.txt)
2. deleted the metaline to ease comparing with my file: [temp_pwkvn_2 (CDSL).txt](https://github.com/sanskrit-lexicon/PWK/files/14372268/temp_pwkvn_2.CDSL.txt)

And the corresponding file from my side: [temp_pwkvn_2 (AB).txt](https://github.com/sanskrit-lexicon/PWK/files/14372277/temp_pwkvn_2.AB.txt)
[Pl. note that my file does not contain the trailing `<info(.*)>` tag.]

After "incorporating" necessary corrections in my file, there are 450 differences in the VN portion with the CDSL file

- some of these belong to the header portion that need to be carried into the metaline
- some are just the relocation of the broken bar, not affecting the metaline, and
- some are within the body portion, not affecting the metaline

Hope @funderburkjim wouldn't be having much issues in **using** my file.

**Comment by Andhrabharati** [comment]
Now, coming to the pw main data, here are **206 header portions** with dhAtu (√) markup--
[dhAtu header lines.txt](https://github.com/sanskrit-lexicon/PWK/files/14375208/dhAtu.header.lines.txt)

Hope, this is convenient enough to be "used" by Jim.
`-----------------------------`

There still remain about 390 diff. lines, out of which 34 lines contain the _ (underscore) character.

Though most of those could be removed as done by Jim (for slp1 has no scope for confusion of vowel-hiatus [but I wonder if these would all "pass" the round-robin test of conversion to another script like Devanagari or IAST and back!!]), I feel some of them need to be retained as they denote a 'space' character within the HW string.

**Comment by Andhrabharati** [comment]
Another 30 lines have the `<ls n="Chr.">` markup by Jim, **which do not point to Boehtlingk's Chrestomathie at all**; I had marked them with the 〔...〕markup, so that they would be easily traceable for properly tagged to their resp. works.

**Comment by Andhrabharati** [comment]
> Note: I have used the GRA model for deriving extra entries for PW (pw_hwextra.txt) from `<k2>`.
> Currently: 2516 extra headwords from 1589 `<k2>`s.

@funderburkjim 

Would you mind explaining about this 1589 number?
I see a huge number of entries that count to nearly **5-6 times** of this!

**Comment by funderburkjim** [comment]
> temp_pwkvn_2.AB.txt

[temp_change_2_3_04.txt](https://github.com/sanskrit-lexicon/PWK/files/14390467/temp_change_2_3_04.txt)  In these (11) instances, AB removes the √ markup. However, Jim thinks these ARE roots (cf. the German translation), hence is retaining the √ markup. @Andhrabharati  Agree?

The remaining 439 (450 - 11) AB changes are agreed by Jim.  Details in [change_2_3.txt](https://github.com/sanskrit-lexicon/PWK/files/14390539/change_2_3.txt)

They are of 3 types:
* 01: 149 add !√ for denominative roots
* 02: 243 add √ for other roots
* 03: 47 change  ¦ placement.  40 of these also require  changes to metaline.
   * L=207313 ?  Diff in accent only. This impacts interpretation by'althws' program in csl-orig
   * L=208905 ? two identical {#X#} before ¦.  Similarly impacts althws interpretation

-------------------------------------------------------------------------------------
The 205 in dhatu.header.lines remain to be processed by Jim.

> 30 lines have the <ls n="Chr."> markup by Jim

@Andhrabharati  Please provide details along with sample of your markup so cdsl can remove inappropriate markup.

**Comment by Andhrabharati** [comment]
> * L=207313 ?  Diff in accent only. This impacts interpretation by'althws' program in csl-orig

There are more such occurrences throughout the CDSL, incl. the pwk.



> * L=208905 ? two identical {#X#} before ¦.  Similarly impacts althws interpretation

The 2nd word has a typo, that I had missed before!
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/8ce5cb12-b688-429c-bfcf-2653ec154ade)

However, pwkvn-7 & SCH have the entries correctly rendered.
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/6088e72d-0272-4d7a-8317-90edb1886b6e)

`---------------------------------`

Pl. have a look at the [change_2_3 (AB).txt](https://github.com/sanskrit-lexicon/PWK/files/14391794/change_2_3.AB.txt) having 'updated' the corrections in Jim's file.

**Comment by Andhrabharati** [comment]
> > 30 lines have the  markup by Jim
> 
> @Andhrabharati Please provide details along with sample of your markup so cdsl can remove inappropriate markup.

There are 60 places in pwkvn portion (and 16 places in the CDSL pwk itself) that have the 〔...〕, which are to be associated with their proper sources.
`-----------------------------`

Here is how I had worked out the **Chr.** taggings--

Boethlingk's 2nd ed. of Chrestomathie (1877) has been cited in pwk [**that ran into 329 pp. (main text) and does not contain more than 34 lines at any page**], unless Benfey's ed. is specifically mentioned.

- `Chr.">([0-9]+),[0-9]{4}	`12 cases
- `Chr.">([0-9]+),[0-9]{3}	`57 cases
- and then, some cases where the page no. is more than 329 or line no. is 34 or more were manually verified and corrected where needed.

Finally, here are the 30 instances that I had untagged as non-Chr. citations--
[non-Chr citations.txt](https://github.com/sanskrit-lexicon/PWK/files/14391874/non-Chr.citations.txt)

**Comment by Andhrabharati** [comment]
While still at the B.'s Chrestomathie, seen that the entries **aMhas** and **aparvan** have 1,10 as a Chr. citation; but the `aparvan` citation points to **RĀJAN.** (previous ls-work) and not to **Chr.**.

Probably there would be more such instances that need untagging `<ls n="Chr.">`

**Comment by Andhrabharati** [comment]
> The 205 in dhatu.header.lines remain to be processed by Jim.

@funderburkjim 

After you process these and other corrections as listed in the above posts, request you to post the **full file(s)** again [so that I can redo the comparison with my file(s) and go to the next step].

**Comment by Andhrabharati** [comment]
> Just by accident, seen few (<20) instances of 'ks' where 'kz' should've been there within the slp1 strings `{#...#}`. There might be such cases in other CDSL works as well, that need to be identified and corrected.

Thought of looking at MW for such instances and found 17 such!!
[While some of these are to be changed to 'kz', others are with typo errors.]
[mw- bad slp words having 'ks'.txt](https://github.com/sanskrit-lexicon/PWK/files/14393965/mw-.bad.slp.words.having.ks.txt)

@drdhaval2785 
Pl. take a note of this and do the needful, as I do not want Jim to divert from the pwk work for now.

**Comment by drdhaval2785** [comment]
Corrected in csl-orig repository for MW as per above comment and above commit.

**Comment by Andhrabharati** [comment]
> (as I did not populate the k2-field yet).

After a long gap, I have got into some mood to take up big works; the first task I did is to populate the k2-field with comma-separated lists (as applicable).

Now my file has 1567 entries in pwk-vn portion (with 2440 extra words) and 7288 entries in pwk-main portion (with 9782 extra words).

**Comment by funderburkjim** [comment]
That's just what I am working on today!  Not sure how to proceed.
1.  I could send my file WITHOUT resolution of the k2 to broken bar
2.  OR I could continue an independent derivation of k2 from broken bar, and then send that file

What do you suggest?

**Comment by Andhrabharati** [comment]
Option 2.

**Comment by gasyoun** [comment]
@funderburkjim how deep are you?

**Comment by Andhrabharati** [comment]
I did the said work in two days; so estimate Jim to take anywhere between 7-10 days (if he is on this work alone), @gasyoun !!

**Comment by funderburkjim** [comment]
[temp_pw_4.zip](https://github.com/sanskrit-lexicon/PWK/files/14485936/temp_pw_4.zip)  has my latest version.

## change files 
relative to my previous post:
* change_2_3.txt  (sections 04-09)
* change_3_4.txt  The k2 markup

I think these changes take into account the various AB mentions,
with exception of the `~1000 Chr. instances having multiple citations together are still not expanded as individual (separate) citations.`
## count comparison:
```
pw body part:   AB has 250+ more than Jim
  7004 entries with k2 multiple headwords  (AB 7288)
  9536 total k2 headwords  (AB 9782)
pw vn part  (AB and Jim about the same)
  1569 entries with k2 multiple headwords  (AB 1567)
  2447 total k2 headwords  (AB 2440)
```

Will be interested to see source of the diffs, esp. in pw body part.

**Comment by Andhrabharati** [comment]
I did **another round of revision** in past two days and the current statistics are--

**pwk-vn portion**
- 1562 entries in  (with 2436 extra words)
- 524 √ items and 1 !√ item

**pwk-main portion**
- 7297 entries in  (with 9719 extra words)
- 3212 √ items and 10 !√ items

Shall look at the differences now with Jim's latest version and post the details.

**Comment by Andhrabharati** [comment]
Starting with the smaller portions--

[pwkvn_4 differences-1 (metalines and header lines).txt](https://github.com/sanskrit-lexicon/PWK/files/14494445/pwkvn_4.differences-1.metalines.and.header.lines.txt)

**Here is the supporting document for the L-205237**
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/0671da02-d1bf-47e8-9d4f-d38d46cdff7b)
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/81cfa66c-2592-42f2-846d-4e986e9c3dec)

**Comment by Andhrabharati** [comment]
It may be noted that I had introduced a new kind of markup of the filled-up portions at the entries.

Also I had separated the portions where a upasarga is clubbed together with a derived word (of the prefixed dhAtu), and also the multiple entries together were separated as individual entries within {#...#}.

And, then I had marked the prefix portions under the dhAtu entries with a `<div n="p">` tag, and pushed them to new lines (such 'new' lines count to 433); this is done in the pwkvn portion to bring it into the **same style** as in the pwk main portion.

Many miscellaneous changes (that seemed appropriate) were also done on-the-fly.

**Comment by Andhrabharati** [comment]
And here is how the CDSL file looks with the upasarga splits implemented--

[pwkvn_4 (CDSL) -prefix splits.txt](https://github.com/sanskrit-lexicon/PWK/files/14497998/pwkvn_4.CDSL.-prefix.splits.txt)

**Comment by Andhrabharati** [comment]
If the above are carried in to the CDSL version and revised file is posted, I shall post the next comparison parts.

**Comment by Andhrabharati** [comment]
If I give a brief about the spl. markup introduced for the filled-up portions at the HW level, probably Jim might appreciate my idea and take up necessary action further (as I intended).

**Comment by funderburkjim** [comment]
> pwkvn_4.CDSL.-prefix.splits.txt

I see that this differs from the current pwkvn portion of temp_pw_4.txt only in that it contains 433 new lines
which begin with `<div n="p">`.  
This revision certainly acceptable.
I could post the next version temp_pw_X.txt:
* Option 1: with just these div differences.
* Option 2: with these div differences and also with  changes indicated in files
pwkvn_4.differences-1. -2. and -3, before posting temp_pw_4a.txt.

Which option are you expecting for my next posting?

**Comment by Andhrabharati** [comment]
Option 2.

**Comment by funderburkjim** [comment]
[pwkvn_4_differences_1_notes.txt](https://github.com/sanskrit-lexicon/PWK/files/14502001/pwkvn_4_differences_1_notes.txt)

@Andhrabharati  Please review these 11, where I either have a question, or have deviated from your suggestion.

**Comment by Andhrabharati** [comment]
> pwkvn_4_differences_1_notes.txt

I agree to all others except to these three items below--

**20941 aBiDarmamahAviBAzA  print change in bb-line?**

Yes, as Eitel is given as the source and it has it thus.

**35757 Why 'QuRQikA'? it is feminine of QuRQika**

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/f04e6dd2-a8c6-420c-9edb-4e19da28cabb)
[Both QuRQika and QuRQikA are feminines.]

**66824 NOT DONE: gajaSAstra appears in ACC.**
The pwk2-299-c has thus
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/ae456cef-453a-46df-9d75-1f3c1d1f85d0)

And as Oppert Catalogue is given as the source, it has thus
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/b546e132-4726-4fa9-9ca6-257ef3d3687b)

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/89314b89-d8d6-42a0-a533-6f25c48787c8)

Though ACC does have gajaSAstra, it does not "happen" to be related to Opp. Cat. 1!!
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/bb476604-2e0c-4813-b44a-47ccd6451840)

**Comment by Andhrabharati** [comment]
BTW, I draw your attention to the formatting that I have in my working files (as seen in the snippets above) that hugely helps 'visibly' identifying many **markup errors** in the data.

This feature of the EmEditor is quite useful, which is not present in the other editors (Notepad++ and Textpad) that I use for other purposes.

**Comment by funderburkjim** [comment]
> gajavEdyaSAstra

Your explanation convincing.  Excellent that you have such comprehensive collection of resources to apply to such questions.  Similarly your Sanskrit-Chinese dictionary resource.

> Both QuRQika and QuRQikA are feminines.

Surely QuRQika is not feminine!  I raised this question because it seems  that inclusion of QuRQikA in k2 field is inconsistent with  your exclusion of feminine forms in other cases (such as tripurARikA).  While I went ahead with your exclusion in k2 of such feminine forms as tripurARikA, I think the 'rule' regarding inclusion/exclusion in k2 of such feminine forms is unclear, and its justification remains dubious. However, within the scope of our current objectives,  this issue is relatively minor, so I'm not going to worry about it further.

**Comment by Andhrabharati** [comment]
> Surely QuRQika is not feminine!

I do agree that it is not feminine, as per Skt. Grammar.

But here, pwk is citing Hemachandra's Prakrit Grammar, which does deviate from Skt. Grammar in many points!!

And the pwk entry (in expanded form) itself clearly gave the two forms--

**ढुण्ढिक** //in व्याकरणढुण्ढिक (probably read °ढुण्ढिका)// and **ढुण्ढिका** **f.** in **हैमप्राकृतव्याकरण**.

[Let me see, if I can locate this in the Pischel's edition (1877)]

> I think the 'rule' regarding inclusion/exclusion in k2 of such feminine forms is unclear

I have opted to ignore the feminine forms (full or contracted) inside braces after adjectives, but not if they occur **preceded by und/oder** [eng. and/or] in the main text itself.

**Comment by Andhrabharati** [comment]
> But here, pwk is citing Hemacandra's Prakrit Grammar, which does deviate from Skt. Grammar in many points!!
> 
> And the pwk entry (in expanded form) itself clearly gave the two forms--
> 
> **ढुण्ढिक** //in व्याकरणढुण्ढिक (probably read °ढुण्ढिका)// and **ढुण्ढिका** **f.** //in **हैमप्राकृतढुण्ढिका**.//
> 
> [Let me see, if I can locate this in the Pischel's edition (1877)]

After seeing that Pischel's edition (of Hemacandra's Grammatik) has NO mention of this, I've looked at ACC, which has lead me to Bühler's Report (1877).

And, it has two entries, as under--
![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/7c9cda13-4143-4292-8d44-f69d37e48292)

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/f83cb36a-b9f1-4d20-99b9-20db2f50c024)

As Boethlingk has mentioned (in pwk), this could be a print error in the first entry in Bühler's Report!!
[And this has nothing to do with Prakrit Grammer (as I thought earlier); and is just a Skt. name of a work.]

**Comment by funderburkjim** [comment]
Option 2 work done.  Result is [temp_pw_5.zip](https://github.com/sanskrit-lexicon/PWK/files/14528047/temp_pw_5.zip)

Cf. [pw_5_work/readme.txt](https://github.com/sanskrit-lexicon/PWK/blob/master/pwkissues/issue106/pw_5_work/readme.txt) starting at `Constructed from ../temp_pw_4a.txt in 3 steps.`

**Comment by Andhrabharati** [comment]
> AB missed these
> `<L>204132<pc>3-256-b<k1>i<k2>3. i ?`
> `<L>205418<pc>4-296-b<k1>i<k2>3. i ?`
> `<L>206831<pc>5-249-c<k1>i<k2>3. i ?`
> `<L>208655<pc>6-298-b<k1>i<k2>3. i  ?`
> `<L>214506<pc>7-320-d<k1>i<k2>3. i  ?`

Yes, I did miss these; though marked the same at L-206831!


> Note: `<L>222590<pc>7-389-d<k1>riktI`
>   Question !√{#riktI#}.  cf. `<L>94190<pc>5-189-b<k1>riktI`

Yes, this is a wrong markup; and there is another such one in pwkvn portion, L-207311. Both these should be without !√.

**Comment by Andhrabharati** [comment]
Here is the last set of differences in pwkvn portion--

[pwkvn_5 differences.txt](https://github.com/sanskrit-lexicon/PWK/files/14529144/pwkvn_5.differences.txt)

**Comment by Andhrabharati** [comment]
Now coming to the pw-main portion.

Noted a total of 983 difference lines between CDSL and AB files.

**Comment by Andhrabharati** [comment]
And request Jim to post the FULL file after necessary corrections are done in the CDSL file, for me to check again.

**Comment by Andhrabharati** [comment]
> Noted a total of 983 difference lines between CDSL and AB files.


> Here is the pw5 diff. file--
> 
> [pw_5 differences latest.txt](https://github.com/sanskrit-lexicon/PWK/files/14529482/pw_5.differences.latest.txt)

Identified that due to some error in "filtering", about 500+ lines got skipped from being listed as difference lines.

To avoid confusion, taking that Jim might've already started looking at my pw_5.differences file, I am not posting a new file.

Many of these differences are in the header lines following the listed corrections in the metalines; and, hope Jim would be identifying those and does the required corrections.

Other differences can be "handled" once Jim finishes working with the present pw_5.differences file.

**Comment by funderburkjim** [comment]
> mostly placement of the '¦'

OK, I'll examine the  '¦' placement while going through the pw_5_diff file.

**Comment by funderburkjim** [comment]
## pw_6

[temp_pw_6.zip](https://github.com/sanskrit-lexicon/PWK/files/14562663/temp_pw_6.zip) the revised pw.txt, version 6

Refer:
* [change_5_6.txt](https://github.com/sanskrit-lexicon/PWK/blob/master/pwkissues/issue106/change_5_6.txt)  the changes from version 5
* [pw_6_work/readme.txt](https://github.com/sanskrit-lexicon/PWK/blob/master/pwkissues/issue106/pw_6_work/readme.txt)
 
## pw5 differences
AB's pw_5.differences.latest.txt file had 3 kinds of 'records'
* metabb  (119) change to both metaline and broken-bar line
* meta      (520) change to metaline only
   * I examined the bbline 
   * 309 matches for "change to bbline yes"   changes made to bbline
   * most of these involve placement of  ¦ in the bbline. AB may disagree with some.
* other  (225) no change to metaline

In pw_6_work/readme.txt, roughly 90 cases noted where Jim disagrees with AB.
There may be a few more such instances that are not noted in the readme.

Is the finish line of this marathon in sight?

**Comment by Andhrabharati** [comment]
> 
> Is the finish line of this marathon in sight?
> 
> 

Certainly yes, I think another three (or four) rounds would finish this (other than expanding the ~1000 Chr. groups); one: resolving the present 90 differences, two:  new differences to be posted by AB as mentioned above, three: Jim's adoptation of the second one and four: final differences (if any) from the third one.

**Comment by Andhrabharati** [comment]
But I can take up the first task only after two days, as I am away from my data and computer now.

Jim may look at other issues in these two days, if he doesn't mind taking a short break from this task/issue.

**Comment by Andhrabharati** [comment]
> <L>124654<pc>7-124-c<k1>sAvairisole<k2>sAva_irisole
>   sAva_irisole -> sAvairisole  slp1 does not need _ for hiatus
>   Note: mw:   sAvaisirole  MW print change?

**;; AB note: MW has erred!!**

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/db0a5d8c-e643-4aa0-a1f5-3159b2d1189c)

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/3edbf075-6b15-43a6-8de5-d4d904f925bb)

**Comment by Andhrabharati** [comment]
Now, coming to the hiatus in slp1.

In normal conditions, it may not be required, as Jim mentioned.
But, when people like AB come-in (**a rare case indeed**), to do a full reading (in a different transliteration, say iast or devanagari) things take a different aspect.

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/b1701a72-8479-40f7-8826-1d6b793251e2)

**Comment by Andhrabharati** [comment]
Here are the ~90 entries with AB comments--

[readme (AB remarks).txt](https://github.com/sanskrit-lexicon/PWK/files/14604587/readme.AB.remarks.txt)

A major portion of these [70+] are so basic errors (which were left in AB's file) that Jim has wondered at a place-- "odd for AB"!

The reason is AB was working with an intention of taking up a full proofing of pwk data (which is recommended by Thomas also) **once this phase is over**, and did a mechanical work mostly in "padding" the alt. HWs (but not  looking at the actual contracted forms having errors).

**Comment by funderburkjim** [comment]
[temp_pw_6a.zip](https://github.com/sanskrit-lexicon/PWK/files/14620112/temp_pw_6a.zip)

Based on review of readme.AB.remarks.txt.
Work is in [pw_6a_work](https://github.com/sanskrit-lexicon/PWK/tree/master/pwkissues/issue106/pw_6a_work).

Only 7 lines of temp_pw_6a.txt differ from temp_pw_6.txt.

@Andhrabharati  Are your next steps:
* revise your version based on comparisons with temp_pw_6a 
* Do a diff between your revised version and  temp_pw_6a

**Comment by Andhrabharati** [comment]
> Work is in [pw_6a_work](https://github.com/sanskrit-lexicon/PWK/tree/master/pwkissues/issue106/pw_6a_work).

> NOTES to AB
> `---`
> ;; And do we revert the few places where the sequence is recently changed to put the items in order?
> JIM: NO, do not make print change to restore alph. order
> `---`

;; AB note: I was referring here to reverting of an entry in L-205237 (**aBiDarma°mahAviBAzA**) earlier relocated to be as per Eitel's (and alphabetical) order.

> `<L>124385<pc>7-121-b<k1>sAradIya`
>  {#nAmamAlA#} -> {#°nAmamAlA#} in bbline  (the ° is needed, but missing in print
> `---`

;; AB note: It is not '°', but '_' that is needed here! 
`<L>124385<pc>7-121-b<k1>sAradIya<k2>sAradIya_nAmamAlA, (SAradIya_nAmamAlA)`

![image](https://github.com/sanskrit-lexicon/PWK/assets/75209130/718d6b98-9b6c-4eb6-a189-5f6f142f9f2a)


> `<L>124169<pc>7-118-a<k1>sAmAnyABAvagranTa<k2>sAmAnyABAvagranTa, ⁅#sAmAnyA⁆°BAvawippanI, ⁅#sAmAnyA⁆°BAvarahasya`
> jim: The '#' is not needed in metaline

;; AB note: I had missed the # earlier and corrected now.

**Comment by Andhrabharati** [comment]
> @Andhrabharati Are your next steps:
> 
>     * revise your version based on comparisons with temp_pw_6a
> 
>     * Do a diff between your revised version and  temp_pw_6a

@funderburkjim 

Noted that you have **just done the 'expansion' of the Chr. entities** [though you mentioned about many other things 'to do', your mind seemed not to deviate from the pwk yet!!]; I would suggest you to integrate (and post) the same so that I can compare the same and we continue further from the same (in a step-by-step manner).

**Comment by Andhrabharati** [comment]
> `<L>63546<pc>4-034-c<k1>parAmarSavAda<k2>parAmarSavAda, parAmarSavAdahetuvicAra, parAmarSavAdArTa`
> 
> AB:   Probably we could use ⁅parAmarSa⁆°hetutAvicAra, and mark it as a print error.
> JIM: Don't make print change now. If <ls>OPP. CAT. 1</ls>. provides proof, then make a change

Oppert catalogue (south India) doesn't contain this item at all.
It is in the Oudh manuscripts catalogue (north-west India), as mentioned in ACC.

And pwk clearly showed the Opp. cat. (as the source) for the two entities on either side, but not for this!!

**Comment by Andhrabharati** [comment]
Here is another (hopefully the final) set of differences in pwkvn portion--
[diff_pwkvn_6a.txt](https://github.com/sanskrit-lexicon/PWK/files/14623109/diff_pwkvn_6a.txt)

**Comment by Andhrabharati** [comment]
A new entry (in pwk main) that got merged into the prev. entry--

439521
(CDSL): `<div n="p">— Mit {#vi#} {%zerfallen, zerbröckeln%}. {#mruc#} {#mro/cati#} {#gatyarTa#}.`

(AB): `<div n="p">— Mit {#vi#} {%zerfallen, zerbröckeln%}.`
(AB): `<LEND>`
(AB): 
(AB): `<L>89274.1<pc>5-113-a<k1>mruc<k2>mruc`
(AB): √{#mruc#}¦, {#mro/cati#} ({#gatyarTa#}).

**Comment by Andhrabharati** [comment]
Here are the first two parts from the differences in pwk main--
[diff_pwk v6a (metalines).txt](https://github.com/sanskrit-lexicon/PWK/files/14623900/diff_pwk.v6a.metalines.txt)

[diff_pwk v6a (addl. upasarga split lines).txt](https://github.com/sanskrit-lexicon/PWK/files/14623903/diff_pwk.v6a.addl.upasarga.split.lines.txt)
**This changes the no. of lines.**

**Comment by Andhrabharati** [comment]
I would like to wait for the implementation of the above posted differences **and also integrating the Chr. expansions** by Jim into pwk_v6a, to post other differences from my file.

**Comment by Andhrabharati** [comment]
I would also like to suggest removing few lines, as we've reached a stage to get rid of them now--

- extra blank lines before L-16301, L-55951, L-69765, L-69947, L-73145 (4 each)
- extra blank lines before <LEND> at L-29488, L-29489 (1 each)
- line no.s 2 and 3 (at the beginning)

**Comment by funderburkjim** [comment]
## versions 7a and 7b

* [temp_pw_7a.zip](https://github.com/sanskrit-lexicon/PWK/files/14640838/temp_pw_7a.zip)  Aims to handle all AB suggested changes since version 6a.
* [temp_pw_7b.zip](https://github.com/sanskrit-lexicon/PWK/files/14640841/temp_pw_7b.zip) applies the 'Chr.' ls expansions to 7a.

Notes are in usual place: [pw_7_work](https://github.com/sanskrit-lexicon/PWK/tree/master/pwkissues/issue106/pw_7_work).

**Comment by gasyoun** [comment]
To watch this duo is mesmerising. 
@Andhrabharati @funderburkjim 
https://youtu.be/AEaA7hhGjCI

**Comment by Andhrabharati** [comment]
Now, we both have the pwkvn portion almost the same.

Started looking at the pwk_main differences without any filters.

- Here are some corrections in the cdsl metalines,with Jim's principle of 'not having' the characters ⁆ and ° (at filling)--

L-2553	(aDyA°lohaka/rRa) -> (aDyAlohaka/rRa)
L-2697	(aDaHprAkSA°yin) -> (aDaHprAkSAyin)
L-5734	anuva°rtita/r⁆ -> anuvartita/r
L-9317	(arGya°pAtra) -> (arGyapAtra)
L-17001	(indu°puzpikA) -> (indupuzpikA)
L-28460	°kuRqamaRqapasaMgraha -> kuRqamaRqapasaMgraha
L-30793	(kISa°romA) -> (kISaromA)
L-36650	AcAryaBadanta°gopadatta -> AcAryaBadantagopadatta
L-37116	gOtamI°nandana⁆ -> gOtamInandana
L-40099	citraSAkApUpaBakza°vikArakriyA -> citraSAkApUpaBakzavikArakriyA
L-43613	(jyO°tsnI) -> (jyOtsnI)
L-50701	dIrGa*°varCikA -> *dIrGavarCikA
L-53543	(drO°Rakajihvi) -> (drORakajihvi)
L-64055	°pariBAzopaskAra -> pariBAzopaskAra
L-64848	(paryu°zaRAzwAhnikA) -> (paryuzaRAzwAhnikA)
L-70023	pOtraM°jIvika⁆ -> pOtraMjIvika
L-70112	(pOrRamAsyADi°karaRa) -> (pOrRamAsyADikaraRa)
L-75465	°PalguRI -> PalguRI
L-75658	(badarI°vanamAhAtmya) -> (badarIvanamAhAtmya)
L-76593	°bahvfcabrAhmaRapaYcikABAzya -> bahvfcabrAhmaRapaYcikABAzya
L-82705	*maDyAntaviBAga°SAstra -> *maDyAntaviBAgaSAstra
L-89573	yajYaM_vo°Qave⁆ -> yajYaM_voQave
L-95910	(lavana°sADikA) -> (lavanasADikA)
L-97438	vajracCedikA_pra°jYApAramitA -> vajracCedikA_prajYApAramitA
L-113109	(SE°rzAyaRa) -> (SErzAyaRa)
L-124366	cintAmaRiH_sA°raRikA -> cintAmaRiH_sAraRikA
L-125542	(sImAnta°dfSvan) -> (sImAntadfSvan)
L-132776	su/a°vas, su/a°vaMs -> su/avas, su/avaMs

- Also noted that while AB version consistently has the (...) entities in the metalines [almost 800 no.s] as per the print, Jim has just about 110 (...) entities.

**Comment by Andhrabharati** [comment]
Here are the **complete** 'unfiltered' difference files in the pwk_main portion--

[diff pwk7b (metalines).txt](https://github.com/sanskrit-lexicon/PWK/files/14651146/diff.pwk7b.metalines.txt) [count: 684]

[diff pwk7b (non-metalines).txt](https://github.com/sanskrit-lexicon/PWK/files/14651147/diff.pwk7b.non-metalines.txt) [count: 1466]

> Now, we both have the pwkvn portion almost the same.

And, here are the **complete** 'unfiltered' difference files in the pwkvn portion--

[diff pwkvn (metalines).txt](https://github.com/sanskrit-lexicon/PWK/files/14651352/diff.pwkvn.metalines.txt) [count: 26]

**Comment by funderburkjim** [comment]
## versions 8 and 8a
Work is in [pw_8_work directory](https://github.com/sanskrit-lexicon/PWK/tree/master/pwkissues/issue106/pw_8_work).

* [temp_pw_8.zip](https://github.com/sanskrit-lexicon/PWK/files/14705385/temp_pw_8.zip)  all changes from AB's 3 files except those in [diff_multiline1.txt](https://github.com/sanskrit-lexicon/PWK/blob/master/pwkissues/issue106/pw_8_work/diff_multiline1.txt)
  * [change_7b_8.txt](https://github.com/sanskrit-lexicon/PWK/blob/master/pwkissues/issue106/change_7b_8.txt) has the changes from previous 7b version.
* [temp_pw_8a.zip](https://github.com/sanskrit-lexicon/PWK/files/14705415/temp_pw_8a.zip) includes the additional diff_multiline1 changes.

Request @Andhrabharati to apply the changes in the two `BEGIN  Jim disagrees with AB for` sections of the pw_8_work/readme.txt file.  If you accept these, then I expect the 8a version will agree with your version.

**Comment by Andhrabharati** [comment]
@funderburkjim 

Would you pl. have a look at this [post](https://github.com/sanskrit-lexicon/PWK/issues/97#issuecomment-1816956126), while I look at the pw_8a file?

**Comment by Andhrabharati** [comment]
Here are the 3 places where AB likes to debate with Jim's opinion.
 

> AB: `<L>124385<pc>7-121-b<k1>sAradIyanAmamAlA<k2>sAradIyanAmamAlA, (SAradIyanAmamAlA)`
> Jim: `<L>124385<pc>7-121-b<k1>sAradIya<k2>sAradIya, (SAradIya), SAradIyanAmamAlA`
>  Note: Also bbline change {#nAmamAlA#} -> {#°nAmamAlA#}  PRINT CHANGE

;; AB remark: There is no "sAradIya" word that occurs in the literature; the word "SAradIya" has already been mentioned at L-111682, and as such there is no need to repeat the same again here.
;; AB remark: It is clearly the suggestion of Boethlingk to consider SAradIya for sAradIya (as a print error in BÜHLER Report) in sAradIyanAmamAlA which is a single word.
;; AB remark: "<lex>Adj.</lex> (<lex>f.</lex> {#A#})" is deleted here in this session, as it appeared redundant here.
;; AB remark: And there is no need to put the ° mark, taking the entry as a single word.
` ----------------------------`

> `{#aBizwipA/si#}¦ <ls>ṚV. 2,20,2</ls> nach <ls>GRASSMANN.</ls> für {#aBi/zwI pAsi#}.`
>   aBizwipA/si -> aBizwipA/(si)  by print

 ;; AB remark: With the adopted norm that the entities having the in-text (...) and [...] be expanded with and without the brackets, this should've been made as an alt. HW group [aBizwipA/, aBizwipA/si].
 ;; AB remark: However, this seems not the intent here; either it should go as just the  "aBizwipA/" as taken by MW, or as "aBizwipA/si" as seen in the ṚV. citation and 'matching' with the GRA emendment.
 ;; AB remark: In either case, this would go as a "print change".
` ----------------------------`

> `{#ISvarItantra#} <lex>n.</lex> und {#ISvare (<ab>Loc.</ab>) nityasuKAvasTApanam#}¦ Titel von Werken.`
>   {#ISvare (<ab>Loc.</ab>) nityasuKAvasTApanam#}  ->
>   {#ISvare#} (<ab>Loc.</ab>) {#nityasuKAvasTApanam#}

;; AB remark: I had felt that the two words (forming the name of the work) need not be separated as individual words, and as such marked thus.

**Comment by Andhrabharati** [comment]
Now, about the slp1 haitus places--
If Jim feels no need for these, in spite of my above [post](https://github.com/sanskrit-lexicon/PWK/issues/106#issuecomment-1996899543), I have no issues in having the hiatus removed at such places.

BTW, there is one place where it is not required, "it does exist" [at L-2991]!

**Comment by Andhrabharati** [comment]
My present version data has additional differences [in non-metalines] in pwk_main (**few: ~150**) and pwkvn (**lot many: ~15k**) portions, probably the comparison could be stopped here.

**Comment by Andhrabharati** [comment]
On a 2nd thought, I have 'modified' both CDSL and AB files a bit; now, the difference line count is **just over 700**.

And, here are the modified files--
[pw (CDSL) 8a.zip](https://github.com/sanskrit-lexicon/PWK/files/14721006/pw.CDSL.8a.zip)
[pw integrated (AB) v1 (for CDSL).zip](https://github.com/sanskrit-lexicon/PWK/files/14721008/pw.integrated.AB.v1.for.CDSL.zip)

**Comment by funderburkjim** [comment]
1. I am unclear on your pw(CDSL)8a version
   * how is it related to the temp_pw_8a version that I uploaded?
3. I suspect  the pw.integrated  version is your latest candidate for final version. Right?
  * how different from your pw(CDSL)8a version ?

Each of these versions has 764942 lines.  and my uploaded temp_pw_8a.txt has 764934 lines --  where do the extra 8 lines in your versions come from?

Also, in the vn section of both your versions, you omit the `<info n="sup_X"/>` field.  This is needed for the displays to show the `[supplement volume X]` note .

**Comment by Andhrabharati** [comment]
> 1. I am unclear on your pw(CDSL)8a version
> * how is it related to the temp_pw_8a version that I uploaded?
> * What use should I make of it?

Yes, it is with some changes done inside.
It can be used to get the diff.s wrt the pw.integrated version; of course your original temp_8a file could also be used, but it will give more (500+) differences.

>     2. I suspect  the pw.integrated  version is your latest candidate for final version. Right?

Yes, for time being.

>     * how different from your pw(CDSL)8a version ?
.
As mentioned above, it has some 700 differences wrt the pw(CDSL)8a

> Each of these versions has 764942 lines. and my uploaded temp_pw_8a.txt has 764934 lines -- where do the extra 8 lines in your versions come from?

I have added extra blank lines after the <H> lines as were at the beginning of the pwkvn file, that were removed in your recent file(s).

> Also, in the vn section of both your versions, you omit the `<info n="sup_X"/>` field. This is needed for the displays to show the `[supplement volume X]` note .

Do you want me to upload the files with the info tags retained as is?

**Comment by funderburkjim** [comment]
> Do you want me to upload the files with the info tags retained as is?

If you can do that readily, then yes.  Otherwise I can find a way to do it.

**Comment by Andhrabharati** [comment]
There are not immediately available; I need to spend a little time to make them.

Probably, it might be better if you do it yourself.

**Comment by funderburkjim** [comment]
Re 'L=124384' --  in your files, you have `{#°nAmamAlA#}`   but you mention `there is no need to put the ° mark`.  

re `{#ISvare (<ab>Loc.</ab>) nityasuKAvasTApanam#}`

This is not proper -- since `<ab>Loc.</ab>` is not Sanskrit. -- 
  The abbreviation needs to be outside of `{#...#}`

**Comment by funderburkjim** [comment]
> be better if you do it yourself.

OK, I'll do that.

**Comment by Andhrabharati** [comment]
> Re 'L=124384' -- in your files, you have `{#°nAmamAlA#}` but you mention `there is no need to put the ° mark`.
> 

My mistake; initially I had reverted my file line as in yours; but  later posted the comments, but not corrected in my file accordingly.

This is what I wanted it to be--
`<L>124385<pc>7-121-b<k1>sAradIyanAmamAlA<k2>⁅sAradIya⁆nAmamAlA, (⁅SAradIya⁆nAmamAlA)`
`{#sAradIya (besser {#SA°#}) nAmamAlA#}¦ <lex>f.</lex> Titel eines Werkes <ls>BÜHLER, Rep. No. 780</ls>.`
`<LEND>`


> re `{#ISvare (<ab>Loc.</ab>) nityasuKAvasTApanam#}`
> 
> This is not proper -- since `<ab>Loc.</ab>` is not Sanskrit. -- The abbreviation needs to be outside of `{#...#}`

So, do we go with the two words separately marked as {#ISvare#} (<ab>Loc.</ab>) {#nityasuKAvasTApanam#}?
[This is not a big point for me to debate upon.]

**Comment by funderburkjim** [comment]
> do we go with the two words separately marked as {#ISvare#} (Loc.) {#nityasuKAvasTApanam#}?

Yes - I can't think of a better solution at the moment.

I've found the extra lines.

That's all my questions for now -- will proceed with analysis/implementation of your changes.

**Comment by funderburkjim** [comment]
Regarding '_' 

You are definitely right that a round-trip of transcoding of X (slp1 -> hk - > slp1) does not result in X when X has certain properties (such as an 'ai' or 'au' hiatus, also 'bh', 'gh' , and maybe a few other cases).  

A similar comment regarding IAST instead of hk.

My view has been that iast and hk should be viewed as faulty and/or incomplete transcoding schemes for Devanagari.  cdsl could take upon itself the task of extending hk and iast to 'remedy' such problems.  But, I have not thought the user reward for such a task is great enough to justify the effort,  since such anomalies are rare.  

While thinking about this, I noticed that the 'simple-search (input=simple)' display needs to be revised so that 'prauga' (MW) yields not only 'prOga' (slp1) but also 'prauga' (slp1).

**Comment by Andhrabharati** [comment]
> My view has been that iast and hk should be viewed as faulty and/or incomplete transcoding schemes for Devanagari.

I've seen that **slp1 itself also** has the drawback of failing in the round-trip conversion, deva - slp1 - deva (or slp1 - deva - slp1) at such places!!

**Comment by funderburkjim** [comment]
## temp_pw_9b.txt
[temp_pw_9b.zip](https://github.com/sanskrit-lexicon/PWK/files/14736625/temp_pw_9b.zip)

This incorporates almost all of AB's latest batch of changes.
See also change_8b_9.txt, change_9_9a.txt and  change_9a_9b.txt for how I analyzed the many different kinds of changes proposed by AB.
See [diff_9b_ab_2.txt](https://github.com/sanskrit-lexicon/PWK/blob/master/pwkissues/issue106/pw_9_work/diff_9b_ab_2.txt) for the differences between temp_pw_9b.txt and AB's *final* file pw.integrated.AB.v1.for.CDSL.txt.

The changes are also integrated into the displays (locally):
<img width="563" alt="image" src="https://github.com/sanskrit-lexicon/PWK/assets/6393033/7659557a-0bf8-429f-a0fc-03199af79940">

@Andhrabharati  When you sign off on temp_pw_9b.txt, I'll install it at Cologne.

**Comment by funderburkjim** [comment]
> I've seen that slp1 itself also has the drawback of failing in the round-trip conversion, deva - slp1 - deva (or slp1 - deva - slp1) at such places!!

I'll believe it when I see it!

I doubt that the Ralph Bunker/Peter Scharf implementation of slp1-deva transcoding has an invertibility problem,  but it may be that my implementation is imperfect.

When (if) you encounter such an instance, open a new issue and provide full details, so I can track down and hopefully correct any such imperfections

**Comment by Andhrabharati** [comment]
> @Andhrabharati When you sign off on temp_pw_9b.txt, I'll install it at Cologne.

Great to see that practically no differences exist between the two versions.

Here are the final changes--
1. While at two entries (L-17562 and L-73947) the hiatus is removed in the header portion, it remained in the metaline.
2. The final form concluded at L-124385 prompted me to look for other places having "(besser" and found 3 entries--
[diff_9b-1.txt](https://github.com/sanskrit-lexicon/PWK/files/14740971/diff_9b-1.txt)
3. The SUrpa°RaKI at L-113882 prompted me to look for other places having "[a-z]°[a-z]" and found 8 lines, out of which 5 are not supposed to be with the ° mark

- 232306	. {#daSa°Sata°#} -> , {#daSa°#}, {#Sata°#}
- 294400	{#nizAda°tva#} -> {#nizAdatva#} ;; print change
- 565127	SUrpa°RaKI -> SUrpaRaKI
- 597908	{#sarvaM°yam#} -> {#sarvaM °yam#}
- 645392	,%} {#pa°da#} -> %}. {#pada°#}

and the remaining 3 lines are the only 'rare' cases having the ° mark within the string (in the digital text; probably there might be few more, which would come out if and when a full proofing takes places to match the file data with the print - i.e. typo errors) [**should we make these changes? if so, what's the best way to do so?**]

- 68425	{#A°nipuRe — dEve#} ;; {#A⁅parvaBaNga⁆nipuRe — dEve#}
- 202051	{#tri°jyotizmatI#} ;; {#tri⁅zwub⁆jyotizmatI#}
- 306317	{#mAMsaM Sva°nipAtitam#} ;; {#mAMsaM Sva⁅daSanAnaNge⁆nipAtitam#}

**Comment by Andhrabharati** [comment]
This is one of the longest sessions that took place-- though at may a times going beyond the "subject matter" (due to my 'uncontrolled' way of corrections!)-- but bringing the text into a good form now.

I would like Jim to think of opening two more issues

- one to tackle the **long-pending** (for over three years now, since I had "promised" to give out my results if the corrections are done in cdsl data as per my proposal) "resolution" of ls-entities; this exercise shall now also include making [the reduced  ls-expansions](https://github.com/sanskrit-lexicon/PWK/issues/96#issuecomment-1646067533) and

- another to "integrate" the vn portion into the main pwk portion, in the same way as done in GRA [this would eliminate the 'pure index entries' (without any "body matter") in the pwk7 vn and retain the 'proper' vn entries having some objective "body matter", and bring the total vn entries count close to what SCH mentioned (14450) from the present 22611 and contain the entries]; I had touched this point (of removing the 'index entries') in the very initial days after the pwkvn got typed by Thomas and added as a new repo as [point 3b](https://github.com/sanskrit-lexicon/PWK/issues/86#issuecomment-1119365151) but it did not get Jim's nod for some reason [**point 3a got corrected in the present session!!**].

I shall take responsibility for these two tasks (the first one does not need much time, and which only I can do [as of now]), but the 2nd one might take a week or so [which Jim could also try out as in GRA initially, and then I had jumped in to give finishing touches jointly].

Look

**Comment by Andhrabharati** [comment]
Finally here is the concluding post from my side at this issue--

> If I give a brief about the spl. markup introduced for the filled-up portions at the HW level, probably Jim might appreciate my idea and take up necessary action further (as I intended).

While in vast majority cases, the "padding" is done at the front of the compound word (as ⁅X⁆°Y), in just 91 cases it is done at the end (as X°⁅Y⁆).

I had presumed that we should somehow have the difference, and thus used the spl. markers '⁅ ⁆';  though the regular '[ ]' could've been used, as it has been used for other purposes in the print, I had thought of having a separate mark to avoid ambiguity.

Jim is requested to recall his opinion on the topic, as note 2 in [L-12291.AB.revised_JF.txt](https://github.com/sanskrit-lexicon/MD/issues/12#issuecomment-1881602678), while I was working at [MD last](https://github.com/sanskrit-lexicon/MD/issues/12#issuecomment-1879952187), wrt the status in MW.]

Now, what use did I have in mind for this marker in practice?

- In case, if (and when) we decide to have the main text (i.e. the header portion) itself changed with the "padded strings" [as done in case of MW], the markers will come in handy.

- We can programmatically match the strings in the k2-field (with marker) and in the following header portion (without marker) and change the header strings easily, and then remove the markers in the k2-field.

**Comment by funderburkjim** [comment]
[temp_pw_9c.zip](https://github.com/sanskrit-lexicon/PWK/files/14749099/temp_pw_9c.zip)  has the few changes mentioned by AB [above](https://github.com/sanskrit-lexicon/PWK/issues/106#issuecomment-2017490352).

[change_9b_9c.txt](https://github.com/sanskrit-lexicon/PWK/blob/master/pwkissues/issue106/change_9b_9c.txt) has the changes.

This version is now installed at Cologne.

Additional revisions of repositories csl-corrections, csl-apidev, hwnorm1 (see commit links above).

The final version changes about 42000 lines out of 764942, or about 5% of lines.
There are now about 12000 'alternate' headwords for pw.
This work has taken about 6 weeks.

Now closing this issue.  Will make a 'placeholder' issue for some additional TODOs.
